src/typechecker/FStar.TypeChecker.Rel.fs

```diff
@@ -1694,9 +1694,10 @@ and solve_rigid_flex_or_flex_rigid_subtyping
 
       let tx = UF.new_transaction () in
       begin
-      match solve_t env eq_prob ({wl' with defer_ok=false; attempting=sub_probs}) with
-      | Success _ ->
+      match solve_t env eq_prob ({wl' with defer_ok=false; wl_implicits = []; attempting=sub_probs}) with
+      | Success (_, imps) ->
          let wl = {wl' with attempting=rest} in
+         let wl = {wl with wl_implicits = wl'.wl_implicits @ imps} in
```

Comment:
This looks good.

Although, FYI, the general structure here is similar to what is provided by `try_solve_without_smt_or_else` (also in this module). I wonder if we should refactor a bit.